src/Registry/TypeRegistry.php

```diff
@@ -933,13 +941,19 @@ protected function prepare_type( string $type_name, $config ) {
 	 *
 	 * @param string $type_name The name of the Type to get from the registry
 	 *
-	 * @return mixed|array<string,mixed>|\GraphQL\Type\Definition\Type|null
+	 * @return ?TypeDef
 	 */
 	public function get_type( string $type_name ) {
 		$key = $this->format_key( $type_name );
 
 		if ( isset( $this->type_loaders[ $key ] ) ) {
-			$type                = $this->type_loaders[ $key ]();
+			$type = $this->type_loaders[ $key ]();
+			/**
+			 * Filter the type before it is loaded into the registry.
+			 *
+			 * @param ?TypeDef $type The type to load.
```

Comment:
This is what's letting us do all this type-narrowing without it being a real breaking change, since `InstrumentSchema::instrument_resolvers() would fatal error if anything other than a `Type` is passed here.
- https://github.com/wp-graphql/wp-graphql/blob/161565b87bd4e3d5bc675ee1f5fbd86801db6493/src/Utils/InstrumentSchema.php#L23

Which has been in the codebase since 2021: https://github.com/wp-graphql/wp-graphql/commit/6c7cc4228e0b6705f5aedf2fd5cf0919d08bcb81

Reply:
@justlevine just to clarify, "type-narrowing" in this case is referring to PHP/PHPStan types and _not_ GraphQL Types?